specs/commandline/plugin.md

```diff
@@ -41,51 +42,144 @@ Aliases:
 
 Aliases:
   install, add
 ```
 
-### notation plugin remove
+### notation plugin upgrade
 
 ```text
-Removes a plugin
+Upgrade a plugin
 
 Usage:
-  notation plugin remove [flags] <plugin name>
+  notation plugin upgrade [flags] <plugin_source>
 
 Flags:
-  -h, --help          help for remove
+  -h, --help                    help for upgrade
+      --checksum string         must match SHA256 of the plugin source                 
+```
+
+### notation plugin uninstall
+
+```text
+Uninstall a plugin
+
+Usage:
+  notation plugin uninstall [flags] <plugin_name>
 
+Flags:
+  -h, --help          help for remove
+  -y, --yes           do not prompt for confirmation
 Aliases:
-  remove, rm, uninstall, delete
+  uninstall, remove, rm
 ```
 
 ## Usage
 
-### Install a plugin
+## Install a plugin 
+
+### Install a plugin from file system
+
+Install a Notation plugin from file system. 
+
+```shell
+$ notation plugin install <file_path>
+```
+
+Upon successful execution, the plugin is copied to Notation's plugin directory. The name and version of the installed plugin is displayed as follows. 
+
+```console
+Successfully installed plugin <plugin name>, version <x.y.z>
+```
+
+If the plugin directory does not exist, it will be created. When an existing plugin is detected, it fails to install and returns the error as follows. Users can use a flag `--force` to skip existence check and force the installation.
+
+```console
+Error: failed to install the plugin, <plugin_name> already installed
+```
+
+If the entered plugin checksum digest doesn't match the published checksum, Notation will return an error message and will not start installation.
+
+```console
+Error: failed to install the plugin, input checksum does not match the published checksum, expected <digest>
+```
+
+### Install a plugin from URL
+
+Install a Notation plugin from a remote shared address and verify the plugin checksum. Notation only supports installing plugins from an HTTPS URL.
 
 ```shell
-notation plugin install <plugin path>
+$ notation plugin install --checksum <digest>
 ```
 
-Upon successful execution, the plugin is copied to plugins directory and name+version of plugin is displayed. If the plugin directory does not exist, it will be created. When an existing plugin is detected, the versions are compared and if the existing plugin is a lower version then it is replaced by the newer version.
+### Install a plugin as an OCI artifact from a registry (for future iteration)
+
+Install a Notation plugin from a registry. Users can verify the plugin's signature with `notation verify` before the plugin installation.
+
+```shell
+$ notation plugin install <registry>/<repository>@<digest>
+```
+
+### Upgrade a plugin to a higher version from file system 
+
+Upgrade a Notation plugin to a higher version from file system and verify the plugin checksum.
+
+```shell
+$ notation plugin upgrade <file_path> --checksum <digest>
```

Comment:
When upgrading from a file system, `--checksum` should be optional? And should we also provide examples when upgrading from URL and OCI registry?

Reply:
Yes. Removed checksum validation here. Added two examples for upgrading plugin from URL and OCI registry